### ISSUE TITLE
Create DisplayName MethodOrderer to order tests by their display name

### ISSUE THREAD

**Issue by dylan-asos** [junit-team/junit5#2212]
In order to sort the tests by the display name attribute attached to tests, create an implementation of `MethodOrderer` that provides this functionality,

## Deliverables

An implementation of `MethodOrderer`, with a sensible fallback, e.g.

```java
public class DisplayNameOrderer implements MethodOrderer {

    public void orderMethods(MethodOrdererContext context) {
        context.getMethodDescriptors().sort(Comparator.comparing(DisplayNameOrderer::getDisplayName));
    }

    private static String getDisplayName(MethodDescriptor descriptor) {
        return descriptor
            .findAnnotation(DisplayName.class)
            .map(DisplayName::value).orElse(""); // Something better than this
    }
}
```

Which can then be used on test classes, as per `@TestMethodOrder(DisplayNameOrderer.class)`.

It's something I've been using where I'm performing some data sequencing tests, i.e. integration rather than unit, but it's been quite handy + satisfies my OCD. 

Happy to do the work + raise a PR if no objections if this was deliberately omitted.

**Comment by sbrannen** [comment]
Tentatively slated for 5.7 M1 for _team discussion_

**Comment by marcphilipp** [comment]
**Team decision:** 
- Add `getDisplayName()` to `MethodDescriptor`
- Add `MethodOrderer.DisplayName` (alphanumeric order is implied)
- Add it to the user guide and release notes

**Comment by marcphilipp** [comment]
@dylan-asos Feel free to go ahead and submit a PR if you're still interested.

**Comment by sbrannen** [comment]
Closed via PR #2220. See https://github.com/junit-team/junit5/pull/2220#issuecomment-605464238 for details.
